types/firestore.d.ts

```diff
@@ -651,7 +706,7 @@ declare namespace FirebaseFirestore {
      * @return An unsubscribe function that can be called to cancel
      * the snapshot listener.
      */
-    onSnapshot(onNext: (snapshot: DocumentSnapshot) => void,
+    onSnapshot(onNext: (snapshot: DocumentSnapshot<T>) => void,
```

Comment:
This is also not exercised  by any tests.

Reply:
Done.